pkg/ck8s/workload_cluster.go

```diff
@@ -242,7 +243,32 @@ func (w *Workload) GetCertificatesExpiryDate(ctx context.Context, machine *clust
 	return response.ExpiryDate, nil
 }
 
-func (w *Workload) RefreshCertificates(ctx context.Context, machine *clusterv1.Machine, nodeToken string, expirationSeconds int, extraSANs []string) (int, error) {
+type ApproveWorkerCSRRequest struct {
+	Seed int `json:"seed"`
+}
+
+type ApproveWorkerCSRResponse struct{}
+
+func (w *Workload) ApproveCertificates(ctx context.Context, machine *clusterv1.Machine, capiToken string, seed int) error {
+	request := ApproveWorkerCSRRequest{}
+	response := &ApproveWorkerCSRResponse{}
+	k8sdProxy, err := w.GetK8sdProxyForControlPlane(ctx, k8sdProxyOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create k8sd proxy: %w", err)
+	}
+
+	header := map[string][]string{
+		"capi-auth-token": {w.authToken},
```

Comment:
I think it might not be a bad idea to either extract this string as a const or have a helper function like `NewHeaderWithAuthToken`. Maybe something for a future improvement tho.